### ISSUE TITLE
Chocolate Fix Question

### ISSUE THREAD

**Issue by Darkosto** [CorgiTaco/BYG#249]
Hello!

I just updated to the latest version (Biomes You Go 1.1.7-BETA-v2 MCV: 1.16.4) and saw there is a new dependency. I have some concerns about this:

1. This is not listed on your mod page so I wasn't sure right away which mod was to be used. Could you list it as a dependency for us to find easily
2. Checking the Chocolate Fix mod page, it seems as though the issues it fixes exist in newer Forge versions, which I am using. I am not sure why I need to download the mod if these problems are fixed in my instance already. Can this be listed as a soft dependency OR if we are running a forge version or later the dependency doesn't need to exist in the mods folder
3. I am bit confused about the inclusion of Chocolate Fix. Is this mod meant to fix an issue when removing Biome mods in an existing world? Isn't it pretty common knowledge that when removing items, blocks, or biomes you may experience some issues in an existing world? I can understand the sentiment but I don't think I want another mod in that pack that really doesn't serve a purpose to me

Just a few thoughts, thanks!

BYG: 1.1.7 Beta v2
MC: 1.16.4
Forge: 35.1.36

**Comment by CorgiTaco** [comment]
1. It is? ![image](https://user-images.githubusercontent.com/66983020/104635916-6dd8cb80-5657-11eb-8beb-302fb76110c8.png)

2. I think itll remain a hard dependency, I explain in 3.

3. It fixes an issue where updating/adding biome mods corrupts existing chunks in existing worlds bc the numerical biome ID's(which change as biome mods are updated/added) are saved only in chunks whereas chocolate fix saves the actual biome string IDs to chunks. As a biome mod dev, alot of users report this issue often and it's a real pain to have to explain to them what occurs and it can get real sad too when people corrupt their worlds altogether bc of it. 

This is also a mojang bug see: https://bugs.mojang.com/browse/MC-202036

And of course here's the scenario where chocolate fix protects against: 
Let's say....
The biome mods I have installed are BYG(Numerical ID's are 100 - 125), BOP(126 - 155), and Prehistoric Fauna(156 - 207).
I create a new world. the biome at 0, 0 is a PHF Hell's Creek.
Only the Biome numerical ID is saved to the chunk so... PHF Hell's Creek in this case is 157.
So biome numerical ID 157 is stored at 0, 0.
Alright... BYG just got an update and they added 3 new biomes. So what happens? Well....
The biome mods I have installed are now BYG(Numerical ID's are 100 - 128), BOP(129 - 158), and Prehistoric Fauna(159 - 210).
So now the biome at 0, 0 still has the numerical ID 157 but.... the ID's have shifted, now the biome at 0, 0 is a Biomes O Plenty Alps Biome.

**Comment by CorgiTaco** [comment]
As me & the dev of Chocolate fix dev understand, this seems to be Lex's stance on doing the fix we care about in Chocolate fix as seen here: https://github.com/MinecraftForge/MinecraftForge/issues/7586 

So as it stands this will be a required dependency unfortunately, until Lex agrees to do such a fix :/

**Comment by Darkosto** [comment]
What I gathered from that issue you linked is a way to deal with new biomes or removing biomes (providing a dummy biome). I don't interpret this to be Lex making a stance against a fix. Especially, after the newer responses from a few hours ago: https://github.com/MinecraftForge/MinecraftForge/issues/7586#issuecomment-767013757 

If the PR made to Forge that was meant to solve this issue isn't working he asks for information to be provided to fix it. It seems like there is a lot of misunderstanding in this situation. Again, my concern is providing a fix for the greater community.

**Comment by alcatrazEscapee** [comment]
Hi @Darkosto 

As the author of the mod in question, I would love it if forge actually fixed the issue present and remove the need for Chocolate. I had submitted an [issue](https://github.com/MinecraftForge/MinecraftForge/issues/7443) back in October, but have basically gotten the response of "we don't think it's an issue" to which I respectfully disagree.

To clarify some points you made,

> Checking the Chocolate Fix mod page, it seems as though the issues it fixes exist in newer Forge versions, which I am using. I am not sure why I need to download the mod if these problems are fixed in my instance already.

The reason chocolate is here is for MC-202036 - which, as per previous comments, is not fixed by Forge yet. (It also fixes MC-197616 incidentally, which is something probably nobody but me cares about). If/when forge decides to actually fix the former, I would also recommend just updating rather than continuing to use chocolate.

> Is this mod meant to fix an issue when removing Biome mods in an existing world?

To an extent. The fact is, when **adding or removing** mods that add biomes, or those mods adding or removing biomes (dummy biomes or not), it can cause biomes in existing worlds to shuffle - something that Forge fixes, mind you, with blocks or items. I have posted a [reproduction and test mod](https://github.com/MinecraftForge/MinecraftForge/issues/7586#issuecomment-767067732) to the linked issue if you feel like investigating that for yourself.
